apps/web/src/components/commands/tokens/Tokens.tsx

```diff
@@ -0,0 +1,67 @@
+import { DataPill } from "../../DataPill";
+import { useQuery } from "react-query";
+import { Tokens } from "./types";
+import { findToken } from "./helpers";
+import Image from "next/image";
+
+export type TokenWidgetProps = {
+  property: "price";
+  chainName: string;
+  token: string;
+  platforms: "native" | "basic";
+};
+
+export const TokenWidget = ({ property, ...props }: TokenWidgetProps) => {
+  switch (property) {
+    case "price":
+      return <TokenPriceWidget {...props} />;
+    default:
+      return null;
+  }
+};
+
+const TokenPriceWidget = ({
+  chainName,
+  token,
+  platforms,
+}: Pick<TokenWidgetProps, "chainName" | "token" | "platforms">) => {
+  const query = useQuery(
+    ["TOKEN", "PRICE", chainName, token, platforms],
+    async () => {
+      if (token.toLowerCase().trim() === "eth") {
+        platforms = "native";
+      }
+      const url = `https://api.portals.fi/v2/tokens?search=${token}&platforms=${platforms}&networks=${chainName}`;
+      const response = await fetch(url);
+      const json = await response.json();
+      const tokenData: Tokens = findToken({
+        query: token,
+        tokenList: json.tokens,
+      });
+
+      return {
+        tokenSymbol: tokenData.symbol,
+        tokenPrice: tokenData.price,
+        tokenImage: tokenData.image,
+        tokenName: tokenData.name,
+      };
+    }
+  );
+
+  if (!query.data) return <div>Loading...</div>;
```

Comment:
This introduces a bug in the case where we get 0 results back, i.e. when token doesn't exist.

Check `query.isLoading` and make a distinction between loading and no data, i.e:
```ts
if (query.isLoading) return <div>Loading...</div>;
if (!query.data) return <div>No data</div>;
```

We might want to think about `query.isError`?

Reply:
@ericzakariasson Does `DataPill` handle `Query` statuses? Seems like it, if so we just need handle `!query.data`